File: apps/desktop/generate-icon.py
#!/usr/bin/env python3
"""
Generate a professional app icon for Nouvelle
Inspired by Notion, Obsidian, and modern app design
"""

from PIL import Image, ImageDraw, ImageFont
import os

def create_icon(size=1024):
    """Create a modern, professional icon for Nouvelle"""

    # Create a new image with transparency
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)

    # Modern color palette - deep indigo/purple gradient
    bg_color1 = (88, 86, 214)      # #5856D6 - vibrant indigo
    bg_color2 = (99, 102, 241)     # #6366F1 - lighter indigo

    # Create rounded square background with gradient effect
    margin = size * 0.12
    corner_radius = size * 0.22

    # Draw background with subtle gradient layers
    for i in range(10):
        offset = i * 2
        alpha = 255 - (i * 10)
        layer_margin = margin + offset

        # Interpolate between colors
        r = int(bg_color1[0] + (bg_color2[0] - bg_color1[0]) * (i / 10))
        g = int(bg_color1[1] + (bg_color2[1] - bg_color1[1]) * (i / 10))
        b = int(bg_color1[2] + (bg_color2[2] - bg_color1[2]) * (i / 10))

        draw.rounded_rectangle(
            [layer_margin, layer_margin, size - layer_margin, size - layer_margin],
            radius=corner_radius,
            fill=(r, g, b, alpha)
        )

    # Draw the stylized "N" logo
    # Using geometric shapes to create a modern "N"
    logo_margin = size * 0.30
    logo_width = size * 0.08

    # Left vertical bar
    draw.rounded_rectangle(
        [logo_margin, logo_margin, logo_margin + logo_width, size - logo_margin],
        radius=logo_width * 0.4,
        fill=(255, 255, 255, 255)
    )

    # Right vertical bar
    draw.rounded_rectangle(
        [size - logo_margin - logo_width, logo_margin, size - logo_margin, size - logo_margin],
        radius=logo_width * 0.4,
        fill=(255, 255, 255, 255)
    )

    # Diagonal bar (creating the N)
    # Calculate points for a rotated rectangle
    top_left = (logo_margin + logo_width * 0.3, logo_margin)
    top_right = (size - logo_margin - logo_width * 0.3, size - logo_margin - logo_width)
    bottom_right = (size - logo_margin, size - logo_margin)
    bottom_left = (logo_margin + logo_width * 1.3, logo_margin + logo_width)

    draw.polygon(
        [top_left, bottom_left, top_right, bottom_right],
        fill=(255, 255, 255, 255)
    )

    return img

def main():
    print("🎨 Generating professional icon for Nouvelle...")

    # Create the master icon at high resolution
    master_icon = create_icon(1024)

    # Save the main icon
    output_dir = "/home/user/new-new-nouvelle/apps/desktop/src-tauri/icons"

    # Save master icon
    master_icon.save(os.path.join(output_dir, "icon.png"))
    print("✓ Created icon.png (1024x1024)")

    # Generate all required sizes
    sizes = {
        "32x32.png": 32,
        "64x64.png": 64,
        "128x128.png": 128,
        "128x128@2x.png": 256,
        "256x256.png": 256,
        "512x512.png": 512,
    }

    for filename, size in sizes.items():
        resized = master_icon.resize((size, size), Image.Resampling.LANCZOS)
        resized.save(os.path.join(output_dir, filename))
        print(f"✓ Created {filename}")

    # Generate Windows icons
    windows_sizes = {
        "Square30x30Logo.png": 30,
        "Square44x44Logo.png": 44,
        "Square71x71Logo.png": 71,
        "Square89x89Logo.png": 89,
        "Square107x107Logo.png": 107,
        "Square142x142Logo.png": 142,
        "Square150x150Logo.png": 150,
        "Square284x284Logo.png": 284,
        "Square310x310Logo.png": 310,
        "StoreLogo.png": 50,
    }

    for filename, size in windows_sizes.items():
        resized = master_icon.resize((size, size), Image.Resampling.LANCZOS)
        resized.save(os.path.join(output_dir, filename))
        print(f"✓ Created {filename}")

    # Save app-icon.png in desktop folder
    master_icon.save("/home/user/new-new-nouvelle/apps/desktop/app-icon.png")
    print("✓ Created app-icon.png")

    print("\n✨ Icon generation complete!")
    print("\nNext steps:")
    print("1. Convert to .icns for macOS: Use 'iconutil' or online converter")
    print("2. Convert to .ico for Windows: Use ImageMagick or online converter")
    print("3. Rebuild your Tauri app to see the new icon!")

if __name__ == "__main__":
    main()

File: apps/desktop/generate-icon.js
#!/usr/bin/env node

/**
 * Generate a professional app icon for Nouvelle
 * Simple, modern design with rounded square and "N" letter
 */

import sharp from "sharp";
import fs from "fs";
import path from "path";
import { fileURLToPath } from "url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

function createIconSVG(size) {
  const cornerRadius = size * 0.225;

  return `
    <svg width="${size}" height="${size}" xmlns="http://www.w3.org/2000/svg">
      <defs>
        <linearGradient id="grad" x1="0%" y1="0%" x2="100%" y2="100%">
          <stop offset="0%" style="stop-color:#5856D6;stop-opacity:1" />
          <stop offset="100%" style="stop-color:#6366F1;stop-opacity:1" />
        </linearGradient>
      </defs>

      <!-- Rounded square background -->
      <rect width="${size}" height="${size}" rx="${cornerRadius}" fill="url(#grad)" />
    </svg>
  `;
}

async function generateIcon(size, outputPath) {
  const svg = createIconSVG(size);
  const buffer = Buffer.from(svg);

  await sharp(buffer).resize(size, size).png().toFile(outputPath);
}

async function main() {
  console.log("🎨 Generating icon for Nouvelle...\n");

  const scriptDir = __dirname;
  const iconsDir = path.join(scriptDir, "src-tauri", "icons");

  // Generate all required sizes
  const sizes = {
    "icon.png": 1024,
    "32x32.png": 32,
    "64x64.png": 64,
    "128x128.png": 128,
    "128x128@2x.png": 256,
    "256x256.png": 256,
    "512x512.png": 512,
  };

  for (const [filename, size] of Object.entries(sizes)) {
    await generateIcon(size, path.join(iconsDir, filename));
    console.log(`✓ Created ${filename}`);
  }

  // Generate Windows icons
  const windowsSizes = {
    "Square30x30Logo.png": 30,
    "Square44x44Logo.png": 44,
    "Square71x71Logo.png": 71,
    "Square89x89Logo.png": 89,
    "Square107x107Logo.png": 107,
    "Square142x142Logo.png": 142,
    "Square150x150Logo.png": 150,
    "Square284x284Logo.png": 284,
    "Square310x310Logo.png": 310,
    "StoreLogo.png": 50,
  };

  for (const [filename, size] of Object.entries(windowsSizes)) {
    await generateIcon(size, path.join(iconsDir, filename));
    console.log(`✓ Created ${filename}`);
  }

  // Generate mobile icons for Android
  const androidSizes = {
    "android/mipmap-mdpi": 48,
    "android/mipmap-hdpi": 72,
    "android/mipmap-xhdpi": 96,
    "android/mipmap-xxhdpi": 144,
    "android/mipmap-xxxhdpi": 192,
  };

  for (const [folder, size] of Object.entries(androidSizes)) {
    const folderPath = path.join(iconsDir, folder);
    if (!fs.existsSync(folderPath)) {
      fs.mkdirSync(folderPath, { recursive: true });
    }

    await generateIcon(size, path.join(folderPath, "ic_launcher.png"));
    await generateIcon(
      size,
      path.join(folderPath, "ic_launcher_foreground.png")
    );
    await generateIcon(size, path.join(folderPath, "ic_launcher_round.png"));
  }
  console.log("✓ Created all Android icons");

  // Generate iOS icons
  const iosSizes = [
    { name: "AppIcon-20x20@1x.png", size: 20 },
    { name: "AppIcon-20x20@2x.png", size: 40 },
    { name: "AppIcon-20x20@2x-1.png", size: 40 },
    { name: "AppIcon-20x20@3x.png", size: 60 },
    { name: "AppIcon-29x29@1x.png", size: 29 },
    { name: "AppIcon-29x29@2x.png", size: 58 },
    { name: "AppIcon-29x29@2x-1.png", size: 58 },
    { name: "AppIcon-29x29@3x.png", size: 87 },
    { name: "AppIcon-40x40@1x.png", size: 40 },
    { name: "AppIcon-40x40@2x.png", size: 80 },
    { name: "AppIcon-40x40@2x-1.png", size: 80 },
    { name: "AppIcon-40x40@3x.png", size: 120 },
    { name: "AppIcon-60x60@2x.png", size: 120 },
    { name: "AppIcon-60x60@3x.png", size: 180 },
    { name: "AppIcon-76x76@1x.png", size: 76 },
    { name: "AppIcon-76x76@2x.png", size: 152 },
    { name: "AppIcon-83.5x83.5@2x.png", size: 167 },
    { name: "AppIcon-512@2x.png", size: 1024 },
  ];

  const iosDir = path.join(iconsDir, "ios");
  if (!fs.existsSync(iosDir)) {
    fs.mkdirSync(iosDir, { recursive: true });
  }

  for (const { name, size } of iosSizes) {
    await generateIcon(size, path.join(iosDir, name));
  }
  console.log("✓ Created all iOS icons");

  // Save app-icon.png
  await generateIcon(1024, path.join(scriptDir, "app-icon.png"));
  console.log("✓ Created app-icon.png");

  // Generate .ico file for Windows (using 256x256 as base)
  const icoSvg = createIconSVG(256);
  const icoBuffer = Buffer.from(icoSvg);
  await sharp(icoBuffer)
    .resize(256, 256)
    .toFile(path.join(iconsDir, "icon.ico"));
  console.log("✓ Created icon.ico");

  console.log("\n✨ Icon generation complete!");
  console.log("\nNext steps:");
  console.log("1. Generate .icns: run the macOS iconutil commands");
  console.log("2. Restart your Tauri app");
}

main().catch(console.error);
